Lib/collections/__init__.py

```diff
@@ -480,15 +492,18 @@ def __getnewargs__(self):
         return _tuple(self)
 
     # Modify function metadata to help with introspection and debugging
-    for method in (
+    methods = (
         __new__,
-        _make.__func__,
+        _make,
         _replace,
         __repr__,
         _asdict,
         __getnewargs__,
-    ):
+    )
+    for method in methods:
         method.__qualname__ = f'{typename}.{method.__name__}'
+    if module is not None:
+        method.__module__ = module
```

Comment:
I wonder if we need to check `module` is a `str` object. In the test, the `module` argument can be passed with `collections` (an instance of `ModuleType`).